docs/kepler_model_server/pipeline.md

```diff
@@ -1,29 +1,20 @@
 # Training Pipeline
 
-As shown in the below figure, a power model can be attributed by the path it was trained. Training pipeline is an abstract of power model training that applies a set of learning methods to a different combination of energy source labels, available metrics, and idle/control plane power isolation choices to each specific group of nodes/machines. 
+Model server serves extensive choices of power models. Training pipeline is an abstract of power model training that applies a set of learning methods to a different combination of energy sources, power isolation methods, available energy-related metrics.  
 
-A training pipeline starts from reading the Kepler-exporting metrics from Prometheus query (prom) and finally submits an archived models to the model database (model DB). 
+## Pipeline
+`pipeline` is composed of three steps, `extract`, `isolate`, and `train`, as shown below. Kepler exports energy-related metrics to the Prometheus in accumulated number over time. `extract` is to convert the aggregated metrics to per-second values and clean up the data separately for each `feature group`. `isolate` is to compute an idle power or control plane power and remove it from the power label of each `energy source`. `train` is to apply each `trainer` to create multiple choices of power models based on the preprocessed data. 
```

Comment:
`Kepler exports energy-related metrics to the Prometheus in accumulated number over time.`
->
`Kepler exports energy-related metrics as Prometheus counter, which provides accumulated number over time.`

Reply:
`extract` is to convert the aggregated metrics to per-second values and clean up the data separately for each `feature group`.
->
The `extract` step is to convert the counter metrics to gauge metrics, similarly to the Prometheus [rate()](https://prometheus.io/docs/prometheus/latest/querying/functions/#rate) function, giving per-second values. The `extract` step also clean up the data separately for each `feature group`.

Reply:
`isolate` is to compute an idle power or control plane power and remove it from the power label of each `energy source`.
->
The power consumption retrieved from the Prometheus query is the absolute power, which is the sum of idle and dynamic power. Additionally, this power is also the total power consumption of all process, including the users' workload, background and OS processes. Then, the `isolate` step can calculate the idle power to isolate the dynamic power consumption.  The `isolate` step can also compute the background and OS process dynamic power consumption and remove it from the power consumption of a given component such as platform, package, dram, etc (i.e., using the `energy source` label).

Reply:
We also have a roadmap to isolate the `activation` power to fully isolate all constant power consumption from the dynamic power.